lib/SIL/SILFunctionBuilder.cpp

```diff
@@ -93,15 +93,25 @@ void SILFunctionBuilder::addFunctionAttributes(SILFunction *F,
     for (auto *A : Attrs.getAttributes<DifferentiableAttr>())
       (void)A->getParameterIndices();
     for (auto *A : Attrs.getAttributes<DifferentiableAttr>()) {
+      auto &ctx = F->getASTContext();
       // Get lowered argument indices.
       auto *paramIndices = A->getParameterIndices();
       assert(paramIndices && "Parameter indices should have been resolved");
       auto *loweredParamIndices = autodiff::getLoweredParameterIndices(
           paramIndices, decl->getInterfaceType()->castTo<AnyFunctionType>());
+      // NOTE(TF-893): Extending capacity is necessary when `origSilFnType` has
```

Comment:
I wonder if the only case that triggers this "capacity extending due to captured variables" logic is the `@differentiable` attribute on [`func original`](https://github.com/apple/swift/blob/tensorflow/stdlib/public/Differentiation/DifferentiationSupport.swift#L218) nested in `func differentiableFunction(from:)`?

Full context: there are known issues regarding differentiation and local variable capture (TF-881). @rxwei mentioned disallowing `@differentiable` attribute on nested functions for now and creating a builtin to support `func differentiableFunction(from:)`. One known user of `differentiableFunction(from:)` is the [custom differentiation tutorial](https://github.com/tensorflow/swift/blob/master/docs/site/tutorials/custom_differentiation.ipynb).

Reply:
Yes, that's the only thing (in the stdlib + swift-apis + tests, at least) that triggers this problem.

It would indeed be nice to remove that and forbid `@differentiable` on nested functions for now.